sycl/source/detail/queue_impl.hpp

```diff
@@ -437,9 +454,9 @@ class queue_impl {
     // Host and interop tasks, however, are not submitted to low-level runtimes
     // and require separate dependency management.
     const CG::CGTYPE Type = Handler.getType();
-    if (MIsInorder && (Type == CG::CGTYPE::CodeplayHostTask ||
-                       Type == CG::CGTYPE::CodeplayInteropTask))
-      Handler.depends_on(MLastEvent);
+    bool NeedSeparateDependencyMgmt =
```

Comment:
This is not used in this function other than passing it to `finalizeHandler`, Please just move it there.